src/CommandLineUtils/Validation/SingleValueValidator.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Nate McMaster.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.ComponentModel.DataAnnotations;
+
+namespace McMaster.Extensions.CommandLineUtils.Validation
+{
+    /// <summary>
+    /// Validates that only a single mapping inside a <see cref="MappedOption{T}"/> was specified on the command-line
+    /// </summary>
+    public class SingleValueValidator : IValidator
+    {
+        private readonly string _errorMessage;
+
+        /// <summary>
+        /// Initializes a new <see cref="SingleValueValidator"/>
+        /// </summary>
+        /// <param name="errorMessage"></param>
+        public SingleValueValidator(string errorMessage = "Only one value is supported.")
+        {
+            _errorMessage = errorMessage;
+        }
+
+        /// <inheritdoc />
+        public ValidationResult GetValidationResult(IOption option, ValidationContext context)
+        {
+            if (option.Values.Count > 1)
```

Comment:
This validator seems to be unnecessary. As user of the API, I would expect this to be enforced already through using `CommandOptionType.SingleValue`.